js/webform_civicrm_payment.js

```diff
@@ -100,7 +100,7 @@ cj(function($) {
   function calculateLineItemAmount() {
     var fieldKey = $(this).data('civicrmFieldKey'),
       amount = getFieldAmount(fieldKey),
-      label = $(this).closest('div.webform-component').find('label').text() || Drupal.t('Contribution'),
+      label = $('label[for=' + this.id + ']').text() || Drupal.t('Contribution'),
```

Comment:
This bit here is to pull the text in the line item table from the label of the field.